src/EnergyPlus/UnitarySystem.cc

```diff
@@ -10902,6 +10937,9 @@ namespace UnitarySystems {
                                                                   state.dataLoopNodes->Node(this->m_SuppCoilAirOutletNode).HumRat));
                 Real64 HCDeltaT = max(0.0, this->DesignMaxOutletTemp - state.dataLoopNodes->Node(this->m_SuppCoilAirInletNode).Temp);
                 Real64 MaxHeatCoilLoad = MDotAir * CpAir * HCDeltaT;
+                if (this->m_NumOfSpeedSuppHeating > 0) {
+                    this->calcMultiStageSuppCoilStageByLoad(state, MaxHeatCoilLoad, FirstHVACIteration);
+                }
                 this->calcUnitarySuppHeatingSystem(state, FirstHVACIteration, SuppPLR, MaxHeatCoilLoad);
```

Comment:
This is the call that bothers me. To call a new supp heater model, and then call the calcUnitarySuppHeatingSystem function to simulate that coil model is redundant. It seems the call to the new coil model can be incorporated into existing calls. This can be resolved later since UnitarySystem needs a major overhaul to clean things up. This is working so well that I hesitate to muck with it now.